# exploratory data analysis<a id='3_Exploratory_Data_Analysis'></a>

## contents
* [introduction](#introduction)
* [imports and load](#imports_and_load)
* [exploration](#exploration_start)
    - [check for bias](#bias_check)
        - [bias in race](#race)
        - [type of diabetes](#type)
    - [demographic visualizations](#demo_viz)


## introduction<a id='introduction'></a>

The goal of this analysis will be to understand this data more fully.  This exploration will inform modelling decisions and possibly reveal insights itself.  This work will include reviewing distributions, scaling, encoding and testing interactions.  Additional data may be merged during exploration as needed and available.

## imports and load<a id='imports_and_load'></a>

In [1]:
import pandas as pd
from numpy import random
import seaborn as sns


Pip requirements for reproducibility

In [2]:
pip list --format=freeze > requirements-2-ja-eda.ipynb.txt

Note: you may need to restart the kernel to use updated packages.


Start with the output of the data wrangling and confirm load

In [3]:
admissions = pd.read_csv('../data/interim/notebook1_output.csv', 
                         index_col='encounter_id')
print(admissions.shape)
admissions.head(3)

(101727, 38)


patient_nbr             race  gender      age admission_type_id  \
encounter_id                                                                    
2278392           8222157        Caucasian  Female   [0-10)             Other   
149190           55629189        Caucasian  Female  [10-20)         emergency   
64410            86047875  AfricanAmerican  Female  [20-30)         emergency   

             discharge_disposition_id admission_source_id  time_in_hospital  \
encounter_id                                                                  
2278392                         Other  physicial referral                 1   
149190                           home      emergency room                 3   
64410                            home      emergency room                 2   

              num_lab_procedures  num_procedures  ...  pioglitazone  \
encounter_id                                      ...                 
2278392                       41               0  ...            No   
149190                        59               0  ...            No   
64410                         11               5  ...            No   

              rosiglitazone  acarbose  miglitol tolazamide insulin  \
encounter_id                                                         
2278392                  No        No        No         No      No   
149190                   No        No        No         No      Up   
64410                    No        No        No         No      No   

             glyburide-metformin  change diabetesMed readmitted  
encounter_id                                                     
2278392                       No      No          No         NO  
149190                        No      Ch         Yes        >30  
64410                         No      No         Yes         NO  

[3 rows x 38 columns]

## exploration<a id='exploration_start'></a>

### check for bias<a id='bias_check'></a>

#### race<a id='race'></a>

As there are demographics present, there is an opportunity to know if any particular group is over or under represented.  Race alone is the first target.  The 2000 [US Census](https://www2.census.gov/library/publications/decennial/2000/briefs/c2kbr01-01.pdf) reports a full population racial mix of 75.1% white and 12.3% black.

The American Diabetes Association [ADA](https://diabetes.org/about-us/statistics/about-diabetes#:~:text=Diabetes%20by%20race%2Fethnicity&text=12.1%25%20of%20non%2DHispanic%20blacks,7.4%25%20of%20non%2DHispanic%20whites) reports that 12.1% of  non-hispanic blacks and 7.4% of non-hispanic whites have diabetes diagnoses.  Ethnicity was not captured in this dataset so cannot be tested.




In [4]:
print('unique race values:', admissions.race.unique())
twoRace = admissions[admissions.race != 'Other']
twoRace_counts = twoRace.race.value_counts()
twoRace_fractions = twoRace.race.value_counts('normalize=True')

adj_white_count = twoRace_counts['Caucasian'] * 7.4
adj_black_count = twoRace_counts['AfricanAmerican'] * 12.1
adj_total = adj_white_count + adj_black_count
adj_white_perc = adj_white_count / adj_total
adj_black_perc = adj_black_count / adj_total
print('expected white fraction:', adj_white_perc)
print('observed white fraction:', twoRace_fractions['Caucasian'])
print('expected black fraction:', adj_black_perc)
print('observed black fraction:', twoRace_fractions['AfricanAmerican'])


unique race values: ['Caucasian' 'AfricanAmerican' 'Other']
expected white fraction: 0.7078017567818881
observed white fraction: 0.798421378564756
expected black fraction: 0.29219824321811194
observed black fraction: 0.201578621435244


To make this dataset unbiased, observations with a race value of "Caucasian" should make up ~71% of all observations.  The number of observations over that can be removed at random.

In [5]:
random.seed(42)
count_black = twoRace.race.value_counts()['AfricanAmerican']
count_white = twoRace.race.value_counts()['Caucasian']
white_to_remove = int(count_white - (count_black / 0.29 - count_black ))
raceWhite = twoRace[twoRace['race'] == 'Caucasian']
raceWhite = raceWhite.sample(raceWhite.shape[0] - white_to_remove)
raceBlack = twoRace[twoRace['race'] == 'AfricanAmerican']
twoRace = pd.concat( [raceBlack, raceWhite])
twoRace.race.value_counts('normalize=True')

Caucasian          0.710004
AfricanAmerican    0.289996
Name: race, dtype: float64

#### type of diabetes<a id='type'></a>

From here, statistics vary widely on prevalance of type 1 and type 2 diabetes by gender and age.  In general, males are more likely to have diabetes than females, but the gap between the sexes is smaller for type 1.  This dataset does contain the first 3 diagnoses for each admission.  Both ICD9 and ICD10 diagnoses codes are used.  ICD9 codes for diabetes start with 250 and ICD10 codes for type 1 start with E10 and type 2 start with E11.

In [7]:
twoRace[~(twoRace['diag_1'].str.contains('250')) &
        ~(twoRace['diag_1'].str.contains('E1')) &
        ~(twoRace['diag_2'].str.contains('250')) & 
        ~(twoRace['diag_2'].str.contains('E1')) &
        ~(twoRace['diag_3'].str.contains('250')) & 
        ~(twoRace['diag_3'].str.contains('E1')) 
       ].shape[0] / twoRace.shape[0] 

0.6218346545866364

Unfortunately, 62% of the observations here don't have any diabetes type diagnoses codes included.  Demographic bias corrections end here.

### demographic visualizations <a id='demo_viz'></a>